modules/flowable-engine-common/src/main/java/org/flowable/common/engine/impl/javax/el/Util.java

```diff
@@ -0,0 +1,741 @@
+// Remove private static CacheKey, CacheValue and PrivilegedGetTccl classes -> They are only relevant to the getExpressionFactory method, and we don't need that one
+// Remove private static getContextLoader method -> Only relevant for the getExpressionFactory method, abd we don't need that one
+// Remove handleThrowable static method -> We are not using it
+// Remove findConstructor static method -> We are not using it
+// Throw fixed error messages instead of using LocalString resource bundle
+// Remove message static method -> ew are not using it
+// Add ExpressionFactory as last method parameter to findMethod, findWrapper, isCoercibleFrom and buildParameters
+// Make public methods accessible if they aren't
+// The findWrapper method has been enhanced with the implementation from https://github.com/eclipse-ee4j/el-ri/blob/4e7c61bce9e7750c2fa6fb85476e33f17b0246b4/api/src/main/java/jakarta/el/ELUtil.java
+// This method follows the JLS more closely and allows picking of ambiguous overloaded methods better.
+// The code in findWrapper is not identical to the one from ELUtil in order to make it more readable for the maintainers of Flowable
+
+// In order for this class to be more easily kept in sync with the Tomcat implementation we should not do style changes, nor fix warnings.
+// Keeping the modifications to minimum would make it easier to keep this class in sync
+class Util {
+
+    private static final Class<?>[] EMPTY_CLASS_ARRAY = new Class<?>[0];
+    private static final Object[] EMPTY_OBJECT_ARRAY = new Object[0];
+
+    /*
+     * This method duplicates code in org.apache.el.util.ReflectionUtil. When
+     * making changes keep the code in sync.
+     */
+    static Method findMethod(Class<?> clazz, Object base, String methodName,
+            Class<?>[] paramTypes, Object[] paramValues, ExpressionFactory factory) {
+
+        if (clazz == null || methodName == null) {
+            throw new MethodNotFoundException("Method not found: " + clazz + "." + methodName + "(" + paramString(paramTypes) + ")");
+        }
+
+        if (paramTypes == null) {
+            paramTypes = getTypesFromValues(paramValues);
+        }
+
+        Method[] methods = clazz.getMethods();
+
+        List<Wrapper<Method>> wrappers = Wrapper.wrap(methods, methodName);
+
+        Wrapper<Method> result = findWrapper(clazz, wrappers, methodName, paramTypes, paramValues, factory);
+
+        return getMethod(clazz, base, result.unWrap());
+    }
+
+    /*
+     * This method duplicates code in org.apache.el.util.ReflectionUtil. When
+     * making changes keep the code in sync.
+     *
+     * Parts of this method and related methods (distributed under the Apache 2.0 License):
+     * Copyright (c) 1997, 2019 Oracle and/or its affiliates and others.
+     * All rights reserved.
+     * Copyright 2004 The Apache Software Foundation
+     */
+    @SuppressWarnings("null")
+    private static <T> Wrapper<T> findWrapper(Class<?> clazz, List<Wrapper<T>> wrappers,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 48. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.